Jenkinsfile

```diff
@@ -0,0 +1,100 @@
+                timeout(5) {
+                    waitUntil {
+                        def r = sh script: 'wget -q http://localhost:3000 -O /dev/null', returnStatus: true
+                        return (r == 0);
+                    }
+                }
+            }
+        }
+    }
+
+
+    stage('Build/Test Host Configurator') {
+        /* Performing some clever trickery to map our ~/.m2 into the container */
+
+        docker.image('maven:3.3-jdk-8').inside(containerArgs) {
+            timestamps {
+                sh 'cd foreman-host-configurator ; mvn -B -U -e -Duser.home=/var/maven -Dmaven.test.failure.ignore=true clean install -DskipTests'
+
+                // let foreman-host-configurator build jar
+                sh 'cd foreman-host-configurator ; rm -f target/foreman-host-configurator.jar'
+                def r = sh script: 'cd foreman-host-configurator ; ./foreman-host-configurator --help', returnStatus: true
+                if (r != 2) {
+                    error('failed to run foreman-host-configurator --help')
+                }
+                // now let it use artifact
+                sh 'cd foreman-host-configurator ; mv target/foreman-host-configurator.jar foreman-host-configurator.jar'
+                r = sh script: 'cd foreman-host-configurator ; ./foreman-host-configurator --help', returnStatus: true
+                if (r != 2) {
+                    error('failed to run foreman-host-configurator --help')
+                }
+            }
+        }
+        def buildArgs = "--build-arg=uid=${uid} --build-arg=gid=${gid} foreman-node-sharing-plugin/src/test/resources/ath-container"
+        docker.build('jenkins/ath', buildArgs)
+        docker.image('jenkins/ath').inside(containerArgs) {
+            sh '''
+            cd foreman-host-configurator ; mvn clean test -Dmaven.test.failure.ignore=true -Duser.home=/var/maven -B
+            '''
+        }
+
+        dir('foreman-host-configurator') {
+            stage('Archive Host Configurator') {
+                junit 'target/surefire-reports/**/*.xml'
+                archive 'foreman-host-configurator'
+                archive 'foreman-host-configurator.jar'
+            }
+        }
+    }
+
+    dir('foreman-node-sharing-plugin') {
+        stage('Build Plugin') {
+            /* Performing some clever trickery to map our ~/.m2 into the container */
+
+            /* Make sure our directory is there, if Docker creates it, it gets owned by 'root' */
+            sh 'mkdir -p $HOME/.m2'
+
+            docker.image('maven:3.3-jdk-8').inside(containerArgs) {
+                timestamps {
+                    sh 'mvn -B -U -e -Dmaven.test.failure.ignore=true -Duser.home=/var/maven clean install -DskipTests'
+                }
+            }
+        }
+
+        stage('Test Plugin') {
+            docker.image('jenkins/ath').inside(containerArgs) {
+                sh '''
+                eval $(./vnc.sh 2> /dev/null)
```

Comment:
Do we really need ATH container with VNC running? I do not see anything browser specific ...